Topics/Tech_Stacks/Unity_ML_Agents.md

```diff
@@ -0,0 +1,199 @@
+
+To make things easier, we should add a reference to both rewards at the start of our agent class
+
+```
+public class CollectRewardAgent : Agent
+{
+    public GameObject rewardSmall;
+    public GameObject rewardLarge;
+    ...
+```
+
+This will allow us to easily access the position of both reward objects as well. To add an observation, we simply go under the CollectObservations function and call ```Sensor.AddObservation()```. We would like to add the x value of all 3 positions, as our agent will only be capable of moving along the x axis. 
+
+```
+public override void CollectObservations(VectorSensor sensor)
+    {
+        sensor.AddObservation(rewardSmall.transform.localPosition.x);
+        sensor.AddObservation(rewardLarge.transform.localPosition.x);
+        sensor.AddObservation(transform.localPosition.x);
+    }
+```
+
+Here we are adding the local positions of all 3 objects. I use local position here so that if we move our training setup around, everything stays functional. However, you can adjust this depending on how your training setup may look like.
+
+Our agent is now able to observe the world around it, as well as make movements left or right. The only thing left to do is to assign a reward function to tell our agent how well it is currently doing.
+# Rewards
+
+Now to assign a reward, this process will differ depending on what you want to train your agent to do. For our scenario, it is sufficient to give the agent a reward when it touches a reward. We will accomplish this with Unity's ```OnTriggerEnter()``` function.
+
+```
+public void OnTriggerEnter(Collider other)
+    {
+        if (other.gameObject == rewardSmall)
+        {
+            SetReward(1f);
+        }
+        if (other.gameObject == rewardLarge)
+        {
+            SetReward(10f);       
+        }
+        EndEpisode();
+    }
+```
+
+From anywhere in Unity, we are able to assign rewards to our agent using the functions ```SetReward()``` and ```AddReward()```. For our purposes, we will set a reward when the agent collides with another object, in this case, either the small or large reward. While the amount you assign to each reward is arbitrary, they should be similar relative to each other. If the rewards are too close to each other, the agent might not learn effectively. Play with your values to find an optimal value.
+
+Something you might have noticed is the ```EndEpisode()``` after we assign the reward. This means that we want our iteration of training to stop when our agent gets its reward, so that it can start training all over again. In order to do so, we have one last function we should call.
+
+```
+public override void OnEpisodeBegin()
+    {
```

Comment:
Same here above, could add a comment maybe

Reply:
Added section explaining episodes and how the OnEpisodeBegin function works